pkg/operator/controller/status_reconciler/status_reconciler.go

```diff
@@ -48,7 +48,10 @@ func (r *StatusReconciler) ReconcileStatus(ctx context.Context, object objectWit
 	if err := r.reconcileCondition(ctx, object, r.getAvailableCondition(ctx, object.GetGeneration())); err != nil {
 		return err
 	}
-	return r.reconcileCondition(ctx, object, r.getDependenciesFulfilledCondition(object.GetGeneration()))
+	if err := r.reconcileCondition(ctx, object, r.getDependenciesFulfilledCondition(object.GetGeneration())); err != nil {
+		return err
+	}
+	return nil
```

Comment:
Whats the difference between this code and the prevoius version (the direct return)?